Invariants

mtao opened the pull request:
Implementation of invariants to replace manually implemented before/after in operations.
Currently the invariant collections are stored in operation settings which is suboptimal, but the current thought is to eventually make them emerge as part of the factory pattern (which will also eventually become responsible for part of the configuration parsing as well)

mtao commented on components/wmtk_components/isotropic_remeshing/internal/IsotropicRemeshing.cpp:
```diff
@@ -19,34 +19,37 @@ IsotropicRemeshing::IsotropicRemeshing(TriMesh& mesh, const double length, const
     {
-        OperationSettings<tri_mesh::EdgeCollapseToMidpoint> op_settings{
-            m_position_handle,
-            m_length_min * m_length_min,
```
@daniel-zint I changed this from max to min, and this results in a failure in the unit tests, but i still believe the code to be wrong due to the choice of variable names. This is because the variable written to is `hackathon/dev` is `max_squared_length` whereas the variable used to assign is `min * min`.
https://github.com/wildmeshing/wildmeshing-toolkit/blob/b27115ad8dde64f3651af39ff8bc53902d00a2ce/src/wmtk/operations/tri_mesh/EdgeCollapseToMidpoint.hpp#L17 .
For this sort of reason I've removed as much of the bracket initialization as possible in the code - and removed default initialization of these configuration things

daniel-zint commented:
There is a change in perspective. The max edge length in isotropic remeshing, is the min edge length for which a split should be performed. I know this is confusing but it's the only thing that makes sense if you look at the classes outside of the context of remeshing. 

> Am 02.09.2023 - 02:19 schrieb Michael Tao:
>
>
> @mtao commented on this pull request.
> 
> In components/wmtk_components/isotropic_remeshing/internal/IsotropicRemeshing.cpp:
> 
> > m_scheduler.add_operation_type<tri_mesh::EdgeSplitAtMidpoint>("split", split_settings); } // collapse { - OperationSettings<tri_mesh::EdgeCollapseToMidpoint> op_settings{ - m_position_handle, - m_length_min * m_length_min, 
> 
> @daniel-zint I changed this from max to min, and this results in a failure in the unit tests, but i still believe the code to be wrong due to the choice of variable names. This is because the variable written to is hackathon/dev is max_squared_length whereas the variable used to assign is min * min.
> https://github.com/wildmeshing/wildmeshing-toolkit/blob/b27115ad8dde64f3651af39ff8bc53902d00a2ce/src/wmtk/operations/tri_mesh/EdgeCollapseToMidpoint.hpp#L17 .
> For this sort of reason I've removed as much of the bracket initialization as possible in the code - and removed default initialization of these configuration things
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.
> Message ID: ***@***.***>
> 
> 

-------------------------------------------------------------------------------------------------
FreeMail powered by mail.de - MEHR SICHERHEIT, SERIOSITÄT UND KOMFORT

mtao commented:
Ok I'll add a comment on this

On Sat, Sep 2, 2023, 05:38 Daniel Zint ***@***.***> wrote:

> There is a change in perspective. The max edge length in isotropic
> remeshing, is the min edge length for which a split should be performed. I
> know this is confusing but it's the only thing that makes sense if you look
> at the classes outside of the context of remeshing.
>
> > Am 02.09.2023 - 02:19 schrieb Michael Tao:
> >
> >
> > @mtao commented on this pull request.
> >
> > In
> components/wmtk_components/isotropic_remeshing/internal/IsotropicRemeshing.cpp:
> >
> > > m_scheduler.add_operation_type<tri_mesh::EdgeSplitAtMidpoint>("split",
> split_settings); } // collapse { -
> OperationSettings<tri_mesh::EdgeCollapseToMidpoint> op_settings{ -
> m_position_handle, - m_length_min * m_length_min,
> >
> > @daniel-zint I changed this from max to min, and this results in a
> failure in the unit tests, but i still believe the code to be wrong due to
> the choice of variable names. This is because the variable written to is
> hackathon/dev is max_squared_length whereas the variable used to assign is
> min * min.
> >
> https://github.com/wildmeshing/wildmeshing-toolkit/blob/b27115ad8dde64f3651af39ff8bc53902d00a2ce/src/wmtk/operations/tri_mesh/EdgeCollapseToMidpoint.hpp#L17
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_wildmeshing_wildmeshing-2Dtoolkit_blob_b27115ad8dde64f3651af39ff8bc53902d00a2ce_src_wmtk_operations_tri-5Fmesh_EdgeCollapseToMidpoint.hpp-23L17&d=DwQFaQ&c=slrrB7dE8n7gBJbeO0g-IQ&r=QoX4ekXM5oUhRWbrXYxsnw&m=r-wqCpwGq_9-_m91hKFByhuG-yaQHE4AR8lWvOLI7kpL4AIVpPa7-N15rpiM8ela&s=Wr9ktfqnZXhuz2DfuXIbPlNkWJmnvxD9hYN0AfzLPYA&e=>
> .
> > For this sort of reason I've removed as much of the bracket
> initialization as possible in the code - and removed default initialization
> of these configuration things
> >
> > —
> > Reply to this email directly, view it on GitHub, or unsubscribe.
> > You are receiving this because you were mentioned.
> > Message ID: ***@***.***>
> >
> >
>
>
> -------------------------------------------------------------------------------------------------
> FreeMail powered by mail.de - MEHR SICHERHEIT, SERIOSITÄT UND KOMFORT
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_wildmeshing_wildmeshing-2Dtoolkit_pull_352-23issuecomment-2D1703779924&d=DwMFaQ&c=slrrB7dE8n7gBJbeO0g-IQ&r=QoX4ekXM5oUhRWbrXYxsnw&m=r-wqCpwGq_9-_m91hKFByhuG-yaQHE4AR8lWvOLI7kpL4AIVpPa7-N15rpiM8ela&s=SBKOmFpz-GTA-wSAPCLuWXgAlHc0nSq4nmyVIGpqiNQ&e=>,
> or unsubscribe
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AAGD6VQG3EAF5UNZVBZI3Z3XYL47VANCNFSM6AAAAAA4DCTPAU&d=DwMFaQ&c=slrrB7dE8n7gBJbeO0g-IQ&r=QoX4ekXM5oUhRWbrXYxsnw&m=r-wqCpwGq_9-_m91hKFByhuG-yaQHE4AR8lWvOLI7kpL4AIVpPa7-N15rpiM8ela&s=2uRZ4IvhDFgPdIbvYqKS57K69-sLK3_lKjwCUd6XBdI&e=>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
